src/service/pom.rs

```diff
@@ -34,79 +37,145 @@ impl crate::server::MyService {
                 Err(err) => return Err(Status::internal(err.to_string())),
             }
         }
-        let path = Path::new(&colink_home)
+        let protocol_home = Path::new(&colink_home)
```

Comment:
please add section comments to briefly describe the goal of the code when your func is >100 LOC.

Reply:
Also, this is not "protocol_home" right? This is for a specific protocol package.
I thought it is the home folder for all protocols, but it is not...